components/script_plugins/unrooted_must_root.rs

```diff
@@ -202,6 +202,11 @@ impl<'a, 'b, 'tcx> UnrootedCx<'a, 'b, 'tcx> {
         // TODO we'll see what's necessary and what is not
         else if match_def_path(cx, did, &["core", "cell", "Ref"])
              || match_def_path(cx, did, &["core", "cell", "RefMut"])
+             || match_def_path(cx, did, &["core", "mem", "size_of"])  // -- is ok?
+             || match_def_path(cx, did, &["malloc_size_of", "MallocSizeOf"])  // -- is ok?
+             || match_def_path(cx, did, &["core", "option", "{{impl}}"])  // -- is ok?
+             || match_def_path(cx, did, &["core", "option", "Option"])  // -- is ok? looks like misuse
```

Comment:
Option is an owning thing, so it shouldn't be here. `Option<MustRootType>` must be rooted. I'm not sure why MallocSizeOf, size_of, and Deref need to be here.

Reply:
That's what I actually thought about Option. I have put all of these on this list, because they needed to be rooted and they come from outside of script component, so I muted the errors in this way and I plan to do similar thing with the remaining errors. There is also another way - by modifying `is_unrooted_ty` to add rules that Option (and other ADTs) are implicitly `#[must_root]` if they contain `#[must_root]` types. I didn't come up with the solution with `is_unrooted_ty` in the first place, and it looks better than current one, so I'll try to move some stuff there.

Reply:
This list is for things that _don't_ need to be rooted even if they have rootables in their type parameters; i.e. reference types and similar.

Reply:
Oh that's fine I guess. Leave a visible comment saying they don't belong 😄